hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -235,15 +236,11 @@ object DefaultSource {
       Option(schema)
     }
 
-    val useNewParquetFileFormat =
-      parameters.getOrElse(
-        USE_NEW_HUDI_PARQUET_FILE_FORMAT.key,
-        USE_NEW_HUDI_PARQUET_FILE_FORMAT.defaultValue).toBoolean &&
-      !metaClient.isMetadataTable &&
-      (globPaths == null || globPaths.isEmpty) &&
-      parameters.getOrElse(REALTIME_MERGE.key(), REALTIME_MERGE.defaultValue())
-        .equalsIgnoreCase(REALTIME_PAYLOAD_COMBINE_OPT_VAL)
-
+    val useNewParquetFileFormat = parameters.getOrDefault(HoodieReaderConfig.FILE_GROUP_READER_ENABLED.key(),
```

Comment:
Looks like `FILE_GROUP_READER_ENABLED` controls whether the `HadoopFsRelation` with new file format is used, which is kind of in a weird state.  Should we keep `hoodie.data source.read.use.new.parquet.file.format` and when it's turned on, the new file group reader is always used for Spark?

Reply:
Discussed offline.  We should keep one config and use `hoodie.file.group.reader.enabled` to control whether new parquet file format is used, along with new file group reader, in Spark.  `hoodie.file.group.reader.enabled` should be used for other engines too, to avoid introducing config per engine.